### PR TITLE
feat: find closest chart to check

### PR DESCRIPTION
**Pull request by vvnkr**
@hermansje Opening this for review.

I started extracting the message building from the tests/state into reporter. It could be made more generic using message structures and more formatters, but I think this is an ok first step for this PR.

### CODE REVIEW

File: tests/test_has_equal_chart.py [hunk 1]
```diff
@@ -353,10 +353,49 @@ def test_check_charts(solution_data, trans, correct):
         )
     ],
 )
-def test_check_charts_trans_on_solution(solution_data, trans, correct):
+def test_has_equal_chart_trans_on_solution(solution_data, trans, correct):
+    user_data = deepcopy(solution_data)
+    solution_data = trans(deepcopy(solution_data))
+    # sct_range is irrelevant in charts
```
Comment (hermansje) [inline tests/test_has_equal_chart.py]: Not anymore

File: tests/test_has_equal_chart.py [hunk 2]
```diff
@@ -353,10 +353,49 @@ def test_check_charts(solution_data, trans, correct):
...
+                ),
+            ),
+            True,
+        ),
+    ],
+)
+def test_has_equal_chart_trans_on_other_chart(solution_data, trans, correct):
     user_data = deepcopy(solution_data)
     solution_data = trans(deepcopy(solution_data))
     # sct_range is irrelevant in charts
```
Comment (hermansje) [inline tests/test_has_equal_chart.py]: Not anymore

**Comment by vvnkr** [comment]
@hermansje Opening this for review.

I started extracting the message building from the tests/state into reporter. It could be made more generic using message structures and more formatters, but I think this is an ok first step for this PR.

**Comment by vvnkr** [comment]
@hermansje Thanks! Naming updated. I'll merge once all checks ran.

**Pull request merged**